File: Anya_Seth/Sliding Window/Max Consecutive Ones III.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int start = 0;
        int end = 0;
        int ans = 0;
        int maxLength  = 0;
        int zeroCount = 0;
        while(end < nums.length){
            if(nums[end] == 0){
                zeroCount++;
            }
            while(zeroCount > k){
                if(nums[start] == 0){
                    zeroCount--;
                }
                start++;
            }
            maxLength = end - start + 1;
            ans = Math.max(ans, maxLength);
            end++;
        }
    return ans;
}
}

File: Lakshay Garg/Leetcode/Sliding Window/Max Consecutive Ones III.cpp
class Solution {
public:
    int longestOnes(vector<int>& arr, int k) {
        int n=arr.size();
        int i=0,j=0,flip=0,ans=INT_MIN;
        while(j<n){
            if(arr[j]==1) j++;
            else{
                if(flip<k){
                    flip++;
                    j++;
                }
                else{
                    ans=max(ans,j-i);
                    while(arr[i]==1) i++;
                    i++;
                    j++;
                }
            }
        }
        ans=max(ans,j-i);
        return ans;
    }
};
